qiskit/qobj/pulse_qobj.py

```diff
@@ -0,0 +1,610 @@
+        if hasattr(self, 'header'):
+            out_dict['header'] = self.header.to_dict()
+        return out_dict
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a new PulseQobjExperiment object from a dictionary.
+
+        Args:
+            data (dict): A dictionary for the experiment config
+
+        Returns:
+            PulseQobjExperiment: The object from the input dictionary.
+        """
+        config = None
+        if 'config' in data:
+            config = PulseQobjExperimentConfig.from_dict(data.pop('config'))
+        header = None
+        if 'header' in data:
+            header = QasmQobjExperimentHeader.from_dict(data.pop('header'))
+        instructions = None
+        if 'instructions' in data:
+            instructions = [
+                PulseQobjInstruction.from_dict(
+                    inst) for inst in data.pop('instructions')]
+        return cls(config, header, instructions)
+
+    def __eq__(self, other):
+        if isinstance(other, PulseQobjExperiment):
+            if self.to_dict() == other.to_dict():
+                return True
+        return False
+
+
+class PulseQobjExperimentConfig:
+    """A config for a single Pulse experiment in the qobj."""
+
+    _data = {}
+
+    def __init__(self, qubit_lo_freq=None, meas_lo_freq=None):
+        """Instantiate a PulseQobjExperimentConfig object.
+
+        Args:
+            qubit_lo_freq (list): List of frequencies (as floats) for the qubit
+                driver LO's in GHz.
+            meas_lo_freq (list): List of frequencies (as floats) for the'
+                measurement driver LO's in GHz.
+        """
+        super(PulseQobjExperimentConfig, self).__init__()
+        self._data = {}
+        if qubit_lo_freq is not None:
+            self._data['qubit_lo_freq'] = qubit_lo_freq
+        if meas_lo_freq is not None:
+            self._data['meas_lo_freq'] = meas_lo_freq
+
+    def to_dict(self):
+        """Return a dictionary format representation of the experiment config.
+
+        Returns:
+            dict: The dictionary form of the PulseQobjExperimentConfig.
+        """
+        return self._data
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a new PulseQobjExperimentConfig object from a dictionary.
+
+        Args:
+            data (dict): A dictionary for the experiment config
+
+        Returns:
+            PulseQobjExperimentConfig: The object from the input dictionary.
+        """
+        return cls(**data)
+
+    def __eq__(self, other):
+        if isinstance(other, PulseQobjExperimentConfig):
+            if self.to_dict() == other.to_dict():
+                return True
+        return False
+
+
+class PulseLibraryItem:
+    """An item in a pulse library."""
+
+    def __init__(self, name, samples):
+        """Instantiate a pulse library item.
+
+        Args:
+            name (str): A name for the pulse.
+            samples (list[complex]): A list of complex values defining pulse
```

Comment:
Is a complex numpy array supported here?

Reply:
Yes, it should work fine as a numpy array. I just copied the type information from the marshmallow schemas which was forcing the input to a `list`. But nothing in how samples is used here would preclude passing in a numpy array. I can update the docstring but I'm not sure what I should put instead.